x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/task/OpenJobPersistentTasksExecutor.java

```diff
@@ -180,7 +181,7 @@ public void validate(OpenJobAction.JobParams params, ClusterState clusterState)
         validateJobAndId(jobId, job);
         // If we already know that we can't find an ml node because all ml nodes are running at capacity or
         // simply because there are no ml nodes in the cluster then we fail quickly here:
-        PersistentTasksCustomMetadata.Assignment assignment = getAssignment(params, clusterState);
+        PersistentTasksCustomMetadata.Assignment assignment = getAssignment(params, clusterState.nodes().getAllNodes(), clusterState);
```

Comment:
This is tricky, it is possible that this validation passes as all the possible assigning nodes are shutting down, but we don't catch that. 

The ML team might remove that last validation (awaiting_lazy_assignment) as now it is unreliable.

Reply:
I don't think the way it is now is any worse than it was before:

- Possible new scenario: validation passes because get assignment thinks an ML node is available when it's actually shutting down, hence the job opens but cannot be assigned
- Old scenario this replaces: validation passes because get assignment finds an ML node is available that is about to be shut down (but nobody apart from the operator knows), job gets assigned, then shortly afterwards the node shuts down and the job cannot be assigned

In both scenarios we end up with a job that is open but cannot be assigned because the cluster doesn't have room for it.  And in both scenarios the solution is to add another ML node to the cluster.